### PR TITLE
ENT-11569: Added documentation for mailto, mailfrom, smtpserver and maxmaillines

### PR DESCRIPTION
**Pull request by nickanderson**
Ticket: ENT-11569
Changelog: None

### CODE REVIEW

**Comment by nickanderson** [comment]
Preview of build result here: https://github.com/cfengine/documentation/pull/3266#issuecomment-2083342353

File: MPF.md [hunk 1]
```diff
@@ -792,19 +792,71 @@ allowed values (`LOG_USER`, `LOG_DAEMON`, `LOG_LOCAL0`, `LOG_LOCAL1`,
...
+The address that `cf-execd` should email agent output to. Defaults to `root@$(default:def.domain)`.
+
+This setting can be customized via Augments, for example:
+
+```json
+{
+  "variables": {
+    "default:def.mailto": {
+        "value": "cfengine-maintainers@example.com",
+        "comment": "Email sent from cf-execd when output differs from the prior execution should be delievered to this distribution group for review"
```
Comment (craigcomstock) [inline MPF.md:804]:
```suggestion
        "comment": "When output differs from the prior execution cf-execd will deliver ??? to this Email address for review"
```

What is sent exactly? the diff, the new output, the previous output, both?

File: MPF.md [hunk 2]
```diff
@@ -792,19 +792,71 @@ allowed values (`LOG_USER`, `LOG_DAEMON`, `LOG_LOCAL0`, `LOG_LOCAL1`,
...
-The SMTP server that `cf-execd` should use to send emails.
+The SMTP server that `cf-execd` should use to send emails. Defaults to `localhost`.
+
+This setting can be customized via Augments, for example:
+
+```json
+{
+  "variables": {
+    "default:def.smtpserver": {
+        "value": "cfengine@example.com",
```
Comment (craigcomstock) [inline MPF.md:837]:
```suggestion
        "value": "smtp.example.com",
```

And is this assuming a plain SMTP endpoint not encrypted or requiring user information for authentication?

File: MPF.md [hunk 3]
```diff
@@ -792,19 +792,71 @@ allowed values (`LOG_USER`, `LOG_DAEMON`, `LOG_LOCAL0`, `LOG_LOCAL1`,
...
+        "comment": "The smtp server that should be used when sending email from cf-execd."
+    }
+  }
+}
+```
 
 ### mailmaxlines
 
-The maximumm number of lines of output that `cf-execd` will email.
+The maximum number of lines of output that `cf-execd` will email. Defaults to `30`.
```
Comment (craigcomstock) [inline MPF.md:846]: This is in relation to the difference between previous execution mentioned earlier? Might want to mention this is line of "execution difference report".

**Review by craigcomstock — COMMENTED** [review]
cool, some questions for clarification

File: MPF.md (hunk 1 above)
Comment (nickanderson) [reply MPF.md:804]: It's the standard  cf-execd email behavior.

If the output differs from the previous run, all output is emailed (up to maxmaillines). So, e.g. if you put a report with $(sys.date) in your policy, your going to get all the output from the agent run each time.

Assuming no other output change, if you report "Hello world" you get an email with all output and no further email until the output changes.

File: MPF.md (hunk 2 above)
Comment (nickanderson) [reply MPF.md:837]: yep.

**Pull request merged**